engine/test/test_path.cpp

```diff
@@ -0,0 +1,111 @@
+/* Copyright (C) 2003-2017 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include "gtest/gtest.h"
+
+#include "prefix.h"
+#include "util.h"
+
+struct TestData
+{
+    const char *path;
+    const char *dir;
+    const char *base;
+};
+
+static const TestData kUnixTestData[] =
+{
+    { "/", "/", "" },
+    { "/foo", "/", "foo" },
+    { "/foo/", "/foo", "" },
+    { "/foo/bar", "/foo", "bar" },
+    { "/foo/bar/", "/foo/bar", "" },
+    { "", "", "" },
+    { "foo", "", "foo" },
+    { "foo/", "foo", "" },
+    { "foo/bar", "foo", "bar" },
+    { "foo/bar/", "foo/bar", "" },
+};
+
+static const TestData kWin32TestData[] =
+{
+    { "C:/", "C:/", "" },
+    { "C:/foo", "C:/", "foo" },
+    { "C:/foo/", "C:/foo", "" },
+    { "C:/foo/bar", "C:/foo", "bar" },
+    { "C:/foo/bar/", "C:/foo/bar", "" },
+    { "C:", "C:", "" },
+    { "C:foo", "C:", "foo" },
+    { "C:foo/", "C:foo", "" },
+    { "C:foo/bar", "C:foo", "bar" },
+    { "C:foo/bar/", "C:foo/bar", "" },
+    { "//", "//", "" },
+    { "///", "///", "" },
+    { "////", "///", "" },
+    { "//Foo", "//Foo", "" },
+    { "//Foo/", "//Foo/", "" },
+    { "//Foo/Bar", "//Foo/Bar", "" },
+    { "//Foo/Bar/", "//Foo/Bar", "" },
+    { "//Foo/Bar/Baz", "//Foo/Bar", "Baz" },
+    { "//Foo/Bar/Baz/", "//Foo/Bar/Baz", "" },
+};
+
+TEST(path, split)
+{
+    MCInitialize();
+
+    for(size_t i = 0; i < sizeof(kUnixTestData) / sizeof(kUnixTestData[0]); i++)
+    {
+        MCAutoStringRef t_dir, t_base;
+        ASSERT_TRUE(MCU_path_split(MCSTR(kUnixTestData[i].path), &(&t_dir), &(&t_base)));
+        
+        MCAutoStringRefAsUTF8String t_dir_c, t_base_c;
+        t_dir_c.Lock(*t_dir);
+        t_base_c.Lock(*t_base);
+        ASSERT_STREQ(*t_dir_c, kUnixTestData[i].dir) << "Failed dir extraction for " << kUnixTestData[i].path;
```

Comment:
I recommend using "EXPECT" rather than "ASSERT" here, since you're checking a condition that doesn't prevent the remainder of the tests from running if it's false.